### ISSUE TITLE
Cannot read apocryphal books

### ISSUE THREAD

**Issue by jaakristioja** [bibletime/bibletime#12]
Moved from old bugtracker. Comments:
# Alister.Hood 2014-05-25 04:58:42 EEST

Using Bibletime 2.10.0 with Sword 1.7.3 and KJVA 3.0.1, the books of the Apocrypha are listed, but Bibletime won't actually navigate to them.  If I select one from the dropdown, nothing happens, and if I type the name e.g. Tobit then it navigates somewhere else e.g. Revelation.
# Alister.Hood 2014-05-25 05:04:56 EEST

Same problem with the DRC translation.
# Alister.Hood 2014-05-25 11:50:12 EEST

Hmmm.
I come back a few hours later and it is working.  I haven't rebooted or reinstalled or anything - just messed around with some of the other modules and closed all sword frontends down and come back and restarted Bibletime...
The only thing is that you get funny results when displaying parallel bibles which don't have the apocrypha (they should really be blank, as they are when displaying the OT with a NT-only parallel bible or vice versa).

Shall I change the status to RESOLVED since I can't even reproduce it?
# Jaak Ristioja devs 2014-05-25 15:10:08 EEST

There may be some problems when displaying works in parallel when the books in them don't match, e.g. when one (the first) Bible translation does not have the Apocrypha, but the others do. Maybe this helps to reproduce it.
# Alister.Hood 2014-05-25 19:06:03 EEST

OK, it looks like there are a few issues, which makes it a little more confusing.  Some of them are related to showing bibles in parallel, and others are not:

(1) it seems that apocryphal "additions" are listed as books, but if you select them nothing happens i.e. in KJVA the "additions to Esther", "Prayer of Azariah", "Susanna" and "Bel and the Dragon".
- bug in bibletime or sword?

(2) modules using "Vulgate" versification list books which they don't actually include, e.g. the DRC lists "Prayer of Manasses", "I Esdras", "II Esdras" (aka 3 & 4 Esdras), "Additional Psalm" and "Laodiceans", after Revelation, but if you select them, all you get is a blank chapter 0 verse 0.
- I guess this is a minor issue, but can/should it be worked around?

(3) as mentioned, when displaying bibles in parallel, bibles which don't include the Apocrypha display the first verse of Revelation for every verse, instead of showing blank, as they would with a missing OT or NT.
- I suspect this can be fixed in bibletime, but it is a relatively minor issue.

(4) A similar but different effect to (3), applying to verses instead of books: when displaying e.g. Daniel 13 and 14 in DRC, parallel Bibles display the start of the book of Hosea instead of showing blank.  Even crazier, load up Daniel 3 in the DRC bible, with something else in a parallel view.  Alongside the apocryphal Daniel 3:24-90 you will see Daniel 3:91/24-100/30, followed by Daniel 4 & 5.  If you view the DRC and the KJVA in parallel, should bibletime match up the additions to Daniel in the two versions, even though one translation published them separately (I think...) and the other translation published them together with the (proto)canon?
- issue for bibletime?

(4) If you select I or II Esdras in the DRC, it is blank as noted in (2).  But if you have KJVA as a parallel bible, it displays a header "The First[/Second] Book of Esdras".  So presumably if you had a bible that actually included I/III and II/IV Esdras as I and II Esdras, with the KJVA in parallel that would display I Esdras/Ezra and II Esdras/Nehemiah, which would of course be totally incorrect.
- based on the comment here I'm guessing this is a module bug; the DRC is using the wrong osisID for these two books:
  https://crosswire.org/svn/sword-tools/trunk/versification/basicv11ns/canon_synodal.h
  Unless bibletime is using the book name somewhere where it should use the osisID...

(5) To reproduce my original issue, open from the bookshelf a bible which doesn't include the apocrypha.  Then "Replace" it with a bible which does include the apocrypha.  Then try selecting a book from the apocrypha - it doesn't work.  Now close Bibletime and reopen it.  Try selecting a book from the apocrypha - it works.  "Replace" the bible with one which doesn't include the apocrypha, then "replace" that again with one which does include the apocrypha, and it still works.  So the issue is that bible "windows" (or "views" or whatever you want to call them) in Bibletime do not allow you to view stuff in the apocrypha _if they were initially opened with a bible that doesn't include the apocrypha_
- I guess this is solely a bibletime issue... and maybe a relatively simple fix?

I don't envy anyone trying to address these issues, but it would be nice to at least fix 4 and 5, and display blanks in parallel bibles whenever they don't contain a verse :)
# Alister.Hood 2014-05-25 19:07:09 EEST

Regarding (2) refer to https://crosswire.org/svn/sword-tools/trunk/versification/basicv11ns/canon_vulg.h

**Comment by kalemas** [comment]
I suppose this issue is solved, at least in part of "read apocrypha", i experience inconsistency in parallel view, but it require another issue item.

**Comment by jaakristioja** [comment]
I'm reopening this so that it will be a reminder for me to verify this completely.

**Comment by gholmlund** [comment]
(1) I have tested KJVA and the books mentioned above all display properly
.
(2) Using VulgClementine, I can confirm that the mentioned books display nothing. I also tested this with Xiphos which does not display them. Based on the comment "books which they don't actually include", the problem seems to me to be in the construction of the module. I don't think we should do anything.

(3) Tested parallel display using KJVA and DRC on the left and ESV on the right. The right side showed a short horizontal line where there was no corresponding book in the ESV. This behavior seems right to me.

(4a) Using DRC and ESV in parallel, I confirm the Daniel 13/Hosea issue as described. I also saw something simular to the description for Daniel 3:24-90

(4b) Using DRC and KJVA in parellel, and selecting I Esdras, the "The First Book of Esdras" which is correct. Same for II Esdras.

(5) Replacing ESV with KJVA and selecting apocrypha book does display them correctly.

Summary: Only (4a) is still a problem.
